src/rd.h

```diff
@@ -154,7 +154,7 @@ static RD_INLINE RD_UNUSED char *rd_strndup(const char *s, size_t len) {
 	char *n = strndup(s, len);
 	rd_assert(n);
 #else
-	char *n = (char *)malloc(len + 1);
+	char *n = (char *)rd_malloc(len + 1);
```

Comment:
This isn't great because if the system has strndup() we might be rd_free() that memory with another allocator (if plugged).

Reply:
I was thinking to leave that to the other PR where you can set a different allocator. But calling system malloc is not really an option then, should call into however rd_malloc is called. But normally yes, all function should use the same allocator/free

Reply:
Right, we can fix it in the future :+1: